packages/Overlay/src/Overlay.js

```diff
@@ -0,0 +1,145 @@
+     * fallback element to programmatically receive focus if no other
+     * tabbable elements are found. For example, you may want a popover's
+     * `<div>` to receive focus if the popover's content includes no
+     * tabbable elements. *Make sure the fallback element has a negative
+     * `tabindex` so it can be programmatically focused.*
+     */
+    fallbackFocus: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.func, HTMLElement]),
+
+    /**
+     * Default: `true`. If `false`, when the trap is deactivated,
+     * focus will *not* return to the element that had focus before activation.
+     */
+    returnFocusOnDeactivate: PropTypes.bool,
+
+    /**
+     * By default, focus trap on deactivation will return to the element
+     * that was focused before activation.
+     */
+    setReturnFocus: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.func, HTMLElement]),
+
+    /**
+     * Default: `true`. If `false`, the `Escape` key will not trigger
+     * deactivation of the focus trap. This can be useful if you want
+     * to force the user to make a decision instead of allowing an easy
+     * way out.
+     */
+    escapeDeactivates: PropTypes.bool,
+
+    /**
+     * Default: `false`. If `true`, a click outside the focus trap will
+     * deactivate the focus trap and allow the click event to do its thing.
+     */
+    clickOutsideDeactivates: PropTypes.bool,
+  }),
+};
+
+const defaultProps = {
+  className: null,
+  backdropClassName: null,
+  children: () => {},
+  showBackdrop: true,
+  onClose: () => {},
+  onAfterOpen: () => {},
+  onAfterClose: () => {},
+  focusTrapOptions: {},
+};
+
+const Overlay = props => {
+  const {
+    backdropClassName,
+    isOpen,
+    children,
+    showBackdrop,
+    onClose,
+    onAfterOpen,
+    onAfterClose,
+    focusTrapOptions,
+    ...moreProps
+  } = props;
+
+  function handleTransitionEnter(node) {
+    // https://github.com/reactjs/react-transition-group/blob/6dbadb594c7c2a2f15bc47afc6b4374cfd73c7c0/src/CSSTransition.js#L44
+    // eslint-disable-next-line no-unused-expressions
+    node.scrollTop;
+  }
+
+  const _focusTrapOptions = {
+    fallbackFocus: () => document.createElement("div"),
+    ...focusTrapOptions,
+  };
+
+  function handleEscKey(event) {
+    if (event.key === "Escape" && isOpen) {
+      event.stopPropagation();
+
+      onClose();
+    }
+  }
+
+  return (
+    <>
+      {isOpen && <LockBodyScroll />}
+      <Portal>
+        <Transition
+          mountOnEnter
+          unmountOnExit
+          in={isOpen}
+          timeout={+tokens.overlay.animationDuration}
+          onEnter={handleTransitionEnter}
+          onEntered={onAfterOpen}
+          onExited={onAfterClose}
+        >
+          {state => (
+            <Wrapper {...moreProps} onKeyDown={handleEscKey}>
+              {showBackdrop && <Backdrop className={backdropClassName} state={state} onClick={onClose} />}
```

Comment:
If we add a `data-pka-anchor='overlay.backdrop'` on the `<Backdrop>` then a custom `className` prop on `<Wrapper>` should allow consumers to target both, and we can eliminate the `backdropClassName` prop.

Reply:
Yes, I will add `data-pka-anchor` attr for backdrop, but don't think that we should remove `backdropClassName`.

I think we should avoid having following CSS on consumer side:
```scss
.my-custom-class-for-overlay {
  // some styles for overlay wrapper  

  [data-pka-anchor='overlay.backdrop'] {
    // some styles for overlay backdrop
  }
}
```

```jsx
import styles from './styles.scss';

export default () => (
  <Modal>
    ...
    <Modal.Overlay className={styles.myCustomClassForOverlay} />
    ...
  </Modal>
);
```

When we use `[data-pka-anchor='overlay.backdrop']` we rely on the fact that element with this selector located somewhere inside in `.my-custom-class-for-overlay`. And this sounds like we rely on internal details of `Overlay` implementation.

I think this way is much cleaner:
```scss
.my-custom-class-for-overlay {
  // some styles for overlay wrapper
}

.my-custom-class-for-overlay-backdrop {
  // some styles for overlay backdrop
}
```

```jsx
import styles from './styles.scss';

export default () => (
  <Modal>
    ...
    <Modal.Overlay className={styles.myCustomClassForOverlay} backdropClassName={styles.myCustomClassForOverlayBackdrop} />
    ...
  </Modal>
);
```